pyincore/analyses/pipelinerestoration/pipelinerestoration.py

```diff
@@ -0,0 +1,203 @@
+
+        restoration_results = self.pipeline_restoration_concurrent_future(
+            self.pipeline_restoration_bulk_input, num_workers,
+            inventory_args)
+        self.set_result_csv_data("pipeline_restoration",
+                                 restoration_results, name=self.get_parameter("result_name"))
+        return True
+
+    def pipeline_restoration_concurrent_future(self, function_name,
+                                               parallelism, *args):
+        """Utilizes concurrent.future module.
+
+        Args:
+            function_name (function): The function to be parallelized.
+            parallelism (int): Number of workers in parallelization.
+            *args: All the arguments in order to pass into parameter function_name.
+
+        Returns:
+            list: A list of dictionary with restoration details
+
+        """
+        res_output = []
+        with concurrent.futures.ProcessPoolExecutor(
+                max_workers=parallelism) as executor:
+            for res_ret in executor.map(function_name, *args):
+                res_output.extend(res_ret)
+
+        return res_output
+
+    def pipeline_restoration_bulk_input(self, damage):
+        """Run analysis for pipeline restoration calculation
+
+        Args:
+            damage (obj): An output of pipeline damage with repair rate
+
+        Returns:
+            restoration_results (list): A list of dictionary restoration times and inventory details
+
+        """
+        restoration_results = []
+        num_available_workers = self.get_parameter("num_available_workers")
+
+        restoration_key = self.get_parameter("restoration_key")
+        if restoration_key is None:
+            restoration_key = "Restoration ID Code"
+
+        restoration_sets = self.restorationsvc.match_list_of_dicts(self.get_input_dataset("dfr3_mapping_set"),
+                                                                   damage, restoration_key)
+
+        for dmg in damage:
+            res = self.restoration_time(dmg, num_available_workers, restoration_sets[dmg['guid']])
+            restoration_results.append(res)
+
+        return restoration_results
+
+    @staticmethod
+    def restoration_time(dmg, num_available_workers, restoration_set):
+        """Calculates restoration time for a single pipeline.
+
+        Args:
+            dmg (obj): Pipeline damage analysis output for a single entry.
+            num_available_workers (int): Number of available workers working on the repairs.
+            restoration_set(obj): Restoration curve(s) to be be used
+
+        Returns:
+            dict: A dictionary with id/guid and restoration time, along with some inventory metadata
+        """
+        res_result = collections.OrderedDict()
+
+        if 'guid' in dmg.keys():
+            res_result['guid'] = dmg['guid']
+        else:
+            res_result['guid'] = 'NA'
+
+        res_result['breakrate'] = dmg['breakrate']
```

Comment:
I guess along the same lines as removing hazard_expose, do we actually need to store breakrate and leakrate since they are in the damage file? Storing these again would be inconsistent with what we've done previously. Thoughts @vnarah2 @longshuicy? I'm ok to address this later.

Reply:
My inclination is to include more columns to make it more readable without the need to join with other inputs(dmg result, input inventory etc.). But, if we want to keep it consistent, I'm ok to remove. Please let me know

Reply:
@navarroc thinking about this again, I can see how someone could misinterpret breakrate and leakrate as part of the outputs of this analysis, when they are not. I'm ok to remove them for consistency

Reply:
That's another concern, they might think it's a different leakrate/breakrate. It would probably be better to show in the example notebook (now or in the future) how to combine the dmg + restoration into a single table and visualize it rather than adding extra columns not produced by this analysis.

Reply:
sound good, removed. Now it only has two columns: guid and repair_time

Reply:
this looks good